### ISSUE TITLE
Promise support

### ISSUE THREAD

**Issue by damianobarbati** [n8io/fantasydata-api#12]
Hey, when do you plan to add promise support? Callback style is pretty aged by now and it does not allow 1) promise chaining 2) yield 3) async/await usage.
Dropping in promise support should be as simple as checking if callback was provided, otherwise return a resolving promise with the method result.

**Comment by n8io** [comment]
I agree promises need to be added. Pull requests are always welcome. Honestly, I haven't kept up with the non-breaking updates to their api so there is a lot of work to be done to bring this wrapper into a more-modern architecture. Promises, async/wait, ES6 in general...
